api/src/main/java/io/opentelemetry/trace/attributes/BooleanAttributeSetter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.attributes;
+
+import io.opentelemetry.trace.Span;
+import javax.annotation.concurrent.Immutable;
+
+/** Defines the behavior for a span attribute with boolean values. */
+@Immutable
+public class BooleanAttributeSetter {
+
+  private final String attributeKey;
+
+  /**
+   * Constructs an attribute object.
+   *
+   * @param attributeKey the attribute name/key
+   */
+  public BooleanAttributeSetter(String attributeKey) {
+    super();
+    this.attributeKey = attributeKey;
+  }
+
+  /**
+   * Returns the attribute name.
+   *
+   * @return the attribute map key
+   */
+  public String key() {
+    return attributeKey;
+  }
+
+  /**
+   * Sets the attribute on the provided span.
+   *
+   * @param span the span to add the attribute to
+   * @param value the value for this attribute
+   */
+  public void set(Span span, boolean value) {
+    span.setAttribute(key(), value);
+  }
```

Comment:
Another maybe useful helper would be `setToCurrent(boolean)` allows users to avoid getting the current `Span` :)

Reply:
Hmm, that would kinda explode the interface if we also wanted things like `setParsedOrRawToCurrent`. In Python, there is a special sentinel value for CURRENT_SPAN that is of type Span. That could work for Java too.

We could solve this via a global helper method `curentSpan` that you would `import static`, then you can write `myAttr.set(currentSpan(), myBool)` which is not much longer than `myAttr.setToCurrent(myBool)`. Maybe such a method even exists already.

Reply:
We do have getCurrentSpan() :)

Reply:
So I am fine to not have the setToCurrent(). But I do want the `setParsedOrRaw` to be removed for the moment.

Reply:
What about `trySetParsed`?

Reply:
Both, I think they can be built easily on top of just `Set` for the moment. Adding new methods is easier than removing in terms of backwards compatibility.

Reply:
It's not possible to implement setParsedOrRaw with set, since you can only set integer values but not string values with that.

Reply:
Now that I see the behavior is even more confusing to set a string value using a int attribute setter.

Reply:
Removed requested methods